Adding cgroups to glossary (#15106)
kubernetes/website: content/en/docs/reference/glossary/cgroups.md

---
title: CGroups

> id: cgroups

Reviewer: ```suggestion
id: cgroup
```

(because most glossary entries are singular). It would be OK to change the text as well to describe what a single cgroup is.

date: 2019-06-25
full_link:
short_description: >

>   cgroup (abbreviated from control group) is a Linux kernel feature that limits, accounts for, and isolates the resource usage (CPU, memory, disk I/O, network) for a collection of processes.

Reviewer: ```suggestion
  A group of Linux processes with optional resource isolation, accounting and limits.
```

@ljnaresh do you prefer this short description (which will be used for a tooltip)? If you prefer your original wording, that's OK by me.

---
title: CGroup

> id: cgroup

Reviewer 2: nit: Should the title be `cgroup` instead of `CGroup` or `control group`?
I just checked the [v2 docs](https://www.kernel.org/doc/Documentation/cgroup-v2.txt):

"cgroup" stands for "control group" and is never capitalized.  The
singular form is used to designate the whole feature and also as a
qualifier as in "cgroup controllers".  When explicitly referring to
multiple individual control groups, the plural form "cgroups" is used.

cgroup is a mechanism to organize processes hierarchically and
distribute system resources along the hierarchy in a controlled and
configurable manner.

date: 2019-06-25
full_link:
short_description: >
  cgroup (abbreviated from control group) is a Linux kernel feature that limits, accounts for, and isolates the resource usage (CPU, memory, disk I/O, network) for a collection of processes.

aka:
tags:
- fundamental
---

> cgroup (abbreviated from control group) is a Linux kernel feature that limits, accounts for, and isolates the resource usage (CPU, memory, disk I/O, network) for a collection of processes.

Reviewer 2: What do you think about removing the phrase `(abbreviated from control group)` and instead adding `(control group)` to the glossary title?

Author: I can see cgroup mentioned in docs and as a result, the search in glossary would be for 'cgroup'. This is the reason I would keep the title as cgroup.

Reviewer 2: OK, I think you could still use cgroup in the doc text,
`{{< glossary_tooltip text="cgroup" term_id="cgroup" >}}`.

I realize that this is a minor point, but do you think it makes sense to rename the file to cgroup.md instead of cgroups.md?

> cgroup (abbreviated from control group) is a Linux kernel feature that limits, accounts for, and isolates the resource usage (CPU, memory, disk I/O, network, etc.) for a collection of processes.

Reviewer 2: nit: remove the `, etc.`

date: 2019-06-25
full_link:
short_description: >

>   A group of Linux processes with optional resource isolation, accounting and limits.

Reviewer 2: - Question: Should the file be renamed to `cgroup.md`?
- Is the short description equivalent to this statement from http://man7.org/linux/man-pages/man7/cgroups.7.html:
A cgroup is a collection of processes that are bound to a set of limits or parameters defined via the cgroup filesystem.
- nit picky: Why is `optional` part of the sentence?

Reviewer: Creating a cgroup and imposing a limit are separate operations, so I think it's fine to point out that limits are optional.
Not all cgroup types have limits, eg [freezer](https://www.kernel.org/doc/Documentation/cgroup-v1/freezer-subsystem.txt) doesn't.

Reviewer 2: OK
